test/cases/testdata/graphql/test-graphql-parse-and-verify.yaml

```diff
@@ -28,7 +28,7 @@ cases:
             }
           }
         `
-        q_ast := {"Operations": [{"Name": "", "Operation": "query", "SelectionSet": [{"Alias": "entity", "Definition": {"Description": "", "Name": "entity", "Type": {"NamedType": "_Entity", "NonNull": false}}, "Name": "entity", "ObjectDefinition": {"BuiltIn": false, "Description": "", "Fields": [{"Description": "", "Name": "entity", "Type": {"NamedType": "_Entity", "NonNull": false}}, {"Description": "", "Name": "__schema", "Type": {"NamedType": "__Schema", "NonNull": true}}, {"Arguments": [{"Description": "", "Name": "name", "Type": {"NamedType": "String", "NonNull": true}}], "Description": "", "Name": "__type", "Type": {"NamedType": "__Type", "NonNull": false}}], "Kind": "OBJECT", "Name": "Query"}, "SelectionSet": [{"ObjectDefinition": {"BuiltIn": false, "Description": "", "Kind": "UNION", "Name": "_Entity", "Types": ["Product", "User"]}, "SelectionSet": [{"Alias": "id", "Definition": {"Description": "", "Name": "id", "Type": {"NamedType": "ID", "NonNull": true}}, "Name": "id", "ObjectDefinition": {"BuiltIn": false, "Description": "", "Fields": [{"Description": "", "Name": "id", "Type": {"NamedType": "ID", "NonNull": true}}], "Kind": "OBJECT", "Name": "User"}}], "TypeCondition": "User"}]}]}]}
+        q_ast := {"Operations": [{"Name": "", "Operation": "query", "SelectionSet": [{"Alias": "entity", "Name": "entity", "SelectionSet": [{"SelectionSet": [{"Alias": "id", "Name": "id"}], "TypeCondition": "User"}]}]}]}
```

Comment:
Test cases like this one had to change because they were tethered to the AST output, and that changed in this PR.